### PR TITLE
docs: step by step - part 1

### PR DESCRIPTION
**Pull request by bwanglzu**
see it lively here: https://ft-docs-step-by-step--jina-docs.netlify.app/docs-step-by-step/2_step_by_step/2_1_install/

please review step-by-step section:

1. installation
2. Two basic concepts: Experiment and Run
3. Login to Jina ecosystem
4. Create training and evaluation (Optional) data

I'll create a follow-up PR for the rest of the step-by-step pages.

### CODE REVIEW

File: docs/2_step_by_step/2_2_experiment_and_run.md [hunk 1]
```diff
@@ -1 +1,33 @@
-# Two basic concepts: Experiment and Run
\ No newline at end of file
+# Two basic concepts: Experiment and Run
+
+Finetuner organize your training based on two concepts: **Experiment** and **Run**.
```
Comment (gmastrapas) [inline docs/2_step_by_step/2_2_experiment_and_run.md:3]: organizes

File: docs/2_step_by_step/2_2_experiment_and_run.md [hunk 2]
```diff
@@ -1 +1,33 @@
...
+
+An Experiment is defined as a general purpose machine learning task you're fine-tuning for.
+A Run is a piece of code that performs the Experiment with specific configurations.
+An Experiment contains a list of Runs,
+each with different configurations.
+For example:
+
++ Experiment: Fine-tune transformer on QuoraQA dataset.
```
Comment (gmastrapas) [inline docs/2_step_by_step/2_2_experiment_and_run.md:11]: a transformer

File: docs/2_step_by_step/2_2_experiment_and_run.md [hunk 3]
```diff
@@ -1 +1,33 @@
...
+
+finetuner.fit(
+  ...,
+  experiment_name='quora-qa-finetune',
+  run_name='quora-qa-finetune-bert',
+)
+```
+
+Please note that these two arguments are `Optional`.
+Finetuner will use the current working directory as default `experiment_name`,
```
Comment (gmastrapas) [inline docs/2_step_by_step/2_2_experiment_and_run.md:32]: as a default

File: docs/2_step_by_step/2_3_login_to_jina_ecosystem.md [hunk 4]
```diff
@@ -1 +1,27 @@
-# Login to Jina ecosystem
\ No newline at end of file
+# Login to Jina ecosystem
+
+Since Finetuner leverage cloud resources for Fine-tuning.
```
Comment (gmastrapas) [inline docs/2_step_by_step/2_3_login_to_jina_ecosystem.md:3]: leverages

File: docs/2_step_by_step/2_3_login_to_jina_ecosystem.md [hunk 5]
```diff
@@ -1 +1,27 @@
...
+it is required to login and obtain a token from Jina.
+It is as simple as:
+
+```python
+import finetuner
+
+finetuner.login()
+```
+
+The browser should pop-up with login options,
```
Comment (gmastrapas) [inline docs/2_step_by_step/2_3_login_to_jina_ecosystem.md:14]: A browser window should pop-up

File: docs/2_step_by_step/2_3_login_to_jina_ecosystem.md [hunk 6]
```diff
@@ -1 +1,27 @@
...
+you will see this in your terminal:
```
Comment (gmastrapas) [inline docs/2_step_by_step/2_3_login_to_jina_ecosystem.md:15]: After login you will see this

File: docs/2_step_by_step/2_3_login_to_jina_ecosystem.md [hunk 7]
```diff
@@ -1 +1,27 @@
...
+
+```bash
+🔐 Successfully login to Jina Ecosystem!
+```
+
+```{admonition} Why I need to login?
+:class: hint
+Login is required since Finetuner need to push your `DocumentArray` into the cloud as training data.
```
Comment (gmastrapas) [inline docs/2_step_by_step/2_3_login_to_jina_ecosystem.md:23]: needs

File: docs/2_step_by_step/2_4_create_training_data.md [hunk 8]
```diff
@@ -1 +1,102 @@
-# Create training data
\ No newline at end of file
+# Create training and evaluation (Optional) data
+
+Finetuner accept [docarray](https://docarray.jina.ai/) `DocumentArray` as training data and evaluation data.
```
Comment (gmastrapas) [inline docs/2_step_by_step/2_4_create_training_data.md:3]: accepts

File: docs/2_step_by_step/2_2_experiment_and_run.md [hunk 9]
```diff
@@ -1 +1,33 @@
...
+
+finetuner.fit(
+  ...,
+  experiment_name='quora-qa-finetune',
+  run_name='quora-qa-finetune-bert',
+)
+```
+
+Please note that these two arguments are `Optional`.
+Finetuner will use the current working directory as a default `experiment_name`,
```
Comment (violenil) [inline docs/2_step_by_step/2_2_experiment_and_run.md:32]: If not supplied, Finetuner will use the current ...

File: docs/2_step_by_step/2_3_login_to_jina_ecosystem.md [hunk 10]
```diff
@@ -1 +1,27 @@
-# Login to Jina ecosystem
\ No newline at end of file
+# Login to Jina ecosystem
+
+Since Finetuner leverages cloud resources for Fine-tuning.
+Before starting a fine-tuning job,
+it is required to login and obtain a token from Jina.
```
Comment (violenil) [inline docs/2_step_by_step/2_3_login_to_jina_ecosystem.md:5]: Since Finetuner leverages cloud resources for fine-tuning, you are required to login and obtain a token from Jina before starting a fine-tuning job.

File: docs/2_step_by_step/2_3_login_to_jina_ecosystem.md [hunk 11]
```diff
@@ -1 +1,27 @@
...
+It is as simple as:
+
+```python
+import finetuner
+
+finetuner.login()
+```
+
+A browser window should pop-up with different login options,
```
Comment (violenil) [inline docs/2_step_by_step/2_3_login_to_jina_ecosystem.md:14]: replace "," with "."

File: docs/2_step_by_step/2_4_create_training_data.md [hunk 12]
```diff
@@ -1 +1,102 @@
-# Create training data
\ No newline at end of file
+# Create training and evaluation (Optional) data
+
+Finetuner accepts [docarray](https://docarray.jina.ai/) `DocumentArray` as training data and evaluation data.
+Because Finetuner follows a [supervised-learning](https://en.wikipedia.org/wiki/Supervised_learning) scheme,
+you should assign label to each `Document` inside `DocumentArray` with:
```
Comment (violenil) [inline docs/2_step_by_step/2_4_create_training_data.md:5]: you should assign a label to each `Document` inside your `DocumentArray` as follows:

File: docs/2_step_by_step/2_4_create_training_data.md [hunk 13]
```diff
@@ -1 +1,102 @@
...
+                tags={'finetuner_label': 'shirt-1'}
+            ),
+        ],
+    ),
+])
+```
+````
+
+As shown in the above code blocks.
+For single-modality model fine-tuning, you only need to create a `Document` with `content` and `tags` with the `finetuner_label`.
```
Comment (violenil) [inline docs/2_step_by_step/2_4_create_training_data.md:79]: merge with prev sentence, ie. replace "." with "," and lowercase "for"

File: docs/2_step_by_step/2_2_experiment_and_run.md [hunk 14]
```diff
@@ -1 +1,33 @@
-# Two basic concepts: Experiment and Run
\ No newline at end of file
+# Two basic concepts: Experiment and Run
+
+Finetuner organizes your training based on two concepts: **Experiment** and **Run**.
+
+An Experiment is defined as a general purpose machine learning task you're fine-tuning for.
```
Comment (guenthermi) [inline docs/2_step_by_step/2_2_experiment_and_run.md:5]:
```suggestion
An Experiment defines the machine learning task you're fine-tuning for in general.
```
For me, a `general purpose ML task` would be something different, i.e., `classification` in general

File: docs/2_step_by_step/2_3_login_to_jina_ecosystem.md [hunk 15]
```diff
@@ -1 +1,27 @@
...
+After login you will see this in your terminal:
```
Comment (guenthermi) [inline docs/2_step_by_step/2_3_login_to_jina_ecosystem.md:15]: Maybe it would be helpful to add where the token is stored. Otherwise, it is a bit confusing, since you mentioned there is a token, but it is not clear where it is. Alternatively, I would suggest to just leaf out that there is a token at all.

File: docs/2_step_by_step/2_4_create_training_data.md [hunk 16]
```diff
@@ -1 +1,102 @@
-# Create training data
\ No newline at end of file
+# Create training and evaluation (Optional) data
+
+Finetuner accepts [docarray](https://docarray.jina.ai/) `DocumentArray` as training data and evaluation data.
```
Comment (guenthermi) [inline docs/2_step_by_step/2_4_create_training_data.md:3]:
```suggestion
Finetuner accepts training data and evaluation data in the form of [docarray](https://docarray.jina.ai/) `DocumentArray` objects.
```

File: docs/2_step_by_step/2_4_create_training_data.md [hunk 17]
```diff
@@ -1 +1,102 @@
...
+For cross-modality (text-to-image) fine-tuning with CLIP,
+you should create a root `Document` which wraps two `chunks` with the `image` and `text` modality.
+The image and text form a pair.
+
+```{admonition} CLIP model explained
+:class: hint
+OpenAI CLIP model wraps two models: a vision transformer and a text transformer.
+During fine-tuning, we're optimizing two models in parallel.
+
+At the model saving time, you will discover we are saving two models to your local directory.
```
Comment (guenthermi) [inline docs/2_step_by_step/2_4_create_training_data.md:90]:
```suggestion
At the model saving time, you will discover, we are saving two models to your local directory. 
```

File: docs/2_step_by_step/2_4_create_training_data.md [hunk 18]
```diff
@@ -1 +1,102 @@
...
+
+```{admonition} CLIP model explained
+:class: hint
+OpenAI CLIP model wraps two models: a vision transformer and a text transformer.
+During fine-tuning, we're optimizing two models in parallel.
+
+At the model saving time, you will discover we are saving two models to your local directory. 
+```
+
+If you need to evaluate metrics on a seperate evaluation data,
```
Comment (guenthermi) [inline docs/2_step_by_step/2_4_create_training_data.md:93]:
```suggestion
If you need to evaluate metrics on separate evaluation data,
```

**Review by guenthermi — COMMENTED** [review]
I added some comments. I would also suggest you to process your text with https://www.grammarly.com/

File: docs/2_step_by_step/2_2_experiment_and_run.md (hunk 14 above)
Comment (jemmyshin) [inline docs/2_step_by_step/2_2_experiment_and_run.md:5]: ``Experiment`` instead of Experiment?

File: docs/2_step_by_step/2_2_experiment_and_run.md [hunk 19]
```diff
@@ -1 +1,33 @@
...
+A Run is a piece of code that performs the Experiment with specific configurations.
```
Comment (jemmyshin) [inline docs/2_step_by_step/2_2_experiment_and_run.md:6]: The same of Run.

File: docs/2_step_by_step/2_4_create_training_data.md [hunk 20]
```diff
@@ -1 +1,102 @@
...
+    ),
+])
+```
+````
+
+As shown in the above code blocks.
+For single-modality model fine-tuning, you only need to create a `Document` with `content` and `tags` with the `finetuner_label`.
+
+For cross-modality (text-to-image) fine-tuning with CLIP,
+you should create a root `Document` which wraps two `chunks` with the `image` and `text` modality.
```
Comment (jemmyshin) [inline docs/2_step_by_step/2_4_create_training_data.md:82]: will we use `dataclass` here? Or just two different `chunk`s?

File: docs/2_step_by_step/2_2_experiment_and_run.md (hunk 14 above)
Comment (bwanglzu) [reply docs/2_step_by_step/2_2_experiment_and_run.md:5]: i think when using `Experiment`, it should be a piece of code. emphasis should use bold or it

File: docs/2_step_by_step/2_4_create_training_data.md (hunk 20 above)
Comment (bwanglzu) [reply docs/2_step_by_step/2_4_create_training_data.md:82]: it's tricky, you can not assign modality in dataclass and you can not assign label

**Review by jupyterjazz — APPROVED** [review]
Looks good! Don't have anything to add to the existing suggestions. 
Just one question - I was wondering where we should explain/show other functionality such as listing/deleting/creating experiments and runs. Is the step-by-step section a proper place or should we keep this part simple and show things in the finetuner-in-action part.

**Comment by bwanglzu** [comment]
@jupyterjazz good point, i think it can be automatically generated from docstring as developer reference. On another side, we'll introduce `list_models` in the `choose backbone model` page

**Review by gmastrapas — APPROVED** [review]
LGTM!
